lib/Basics/CrashHandler.cpp

```diff
@@ -470,6 +474,60 @@ void crashHandlerSignalHandler(int signal, siginfo_t* info, void* ucontext) {
 }
 #endif
 
+#ifdef _WIN32
+
+static std::string miniDumpDirectory = "C:\\temp";
+
+// This function is called once after installing the exceptionFilter.
+void setMiniDumpDirectory() { miniDumpDirectory = TRI_GetTempPath(); }
+
+void createMiniDump(EXCEPTION_POINTERS* pointers) {
+  time_t rawtime;
+  time(&rawtime);
+  struct tm timeinfo;
+  _gmtime64_s(&timeinfo, &rawtime);
+  char time[20];
+  strftime(time, sizeof(time), "%Y-%m-%dT%H-%M-%S", &timeinfo);
+  
+  char filename[MAX_PATH];
+  sprintf(filename, "%s\\%s_%d_%d.dmp", miniDumpDirectory.c_str(), time, GetCurrentProcessId(), GetCurrentThreadId());
+  HANDLE hFile = CreateFile(filename, GENERIC_WRITE, 0, nullptr,
+                            CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+
+  if (hFile == INVALID_HANDLE_VALUE) {
+    LOG_TOPIC("ba80e", FATAL, arangodb::Logger::CRASH)
+        << "Could not open minidump file: " << GetLastError();
+    return;
+  }
+
+  MINIDUMP_EXCEPTION_INFORMATION exceptionInfo;
+  exceptionInfo.ThreadId = GetCurrentThreadId();
+  exceptionInfo.ExceptionPointers = pointers;
+  exceptionInfo.ClientPointers = FALSE;
+
+  if (MiniDumpWriteDump(GetCurrentProcess(), GetCurrentProcessId(), hFile,
+                        MINIDUMP_TYPE(MiniDumpNormal | MiniDumpWithProcessThreadData | MiniDumpWithDataSegs),
+                        pointers ? &exceptionInfo : nullptr, nullptr, nullptr)) {
+    LOG_TOPIC("93315", FATAL, arangodb::Logger::CRASH) << "Wrote minidump: " << filename;
+  } else {
+    LOG_TOPIC("af06b", FATAL, arangodb::Logger::CRASH)
+        << "Failed to write minidump: " << GetLastError();
+  }
+
+  CloseHandle(hFile);
+}
+
+LONG CALLBACK unhandledExceptionFilter(EXCEPTION_POINTERS* pointers) {
+  TRI_ASSERT(pointers && pointers->ExceptionRecord);
+  LOG_TOPIC("87ff4", FATAL, arangodb::Logger::CRASH)
```

Comment:
I suggest to avoid dynamic memory allocations during unhandled exception processing. And do the logging like it is done for *nix crash handler. It uses pre-allocated buffers